cmd/snaplock/runinhibit/inhibit.go

```diff
@@ -283,3 +278,115 @@ func RemoveLockFile(snapName string) error {
 
 	return nil
 }
+
+// needed for better mocking
+type ticker interface {
+	Wait()
+}
+
+type timeTicker struct {
+	interval time.Duration
+}
+
+func (t *timeTicker) Wait() {
+	time.Sleep(t.interval)
+}
+
+var newTicker = func(interval time.Duration) ticker {
+	return &timeTicker{interval}
+}
+
+// WaitWhileInhibited blocks until snap is not inhibited anymore and then returns
+// a locked hint file lock.
+//
+// The `inhibited` callback is run if the snap is initially inhibited, otherwise the `notInhibited`
+// callback is run. In either case, this callback runs with the hint file lock held`.
+//
+// If inhibited callback returns true, WaitWhileInhibited returns instantly without
+// waiting for the snap not being inhibited.
+//
+// NOTE: A snap without a hint file is considered not inhibited and a nil FileLock is returned.
+//
+// NOTE: It is the caller's responsibility to release the returned file lock.
+var WaitWhileInhibited = func(snapName string, notInhibited func() error, inhibited func(hint Hint, inhibitInfo *InhibitInfo) (cont bool, err error), interval time.Duration) (flock *osutil.FileLock, retErr error) {
+	ticker := newTicker(interval)
+
+	for {
+		flock, err := osutil.OpenExistingLockForReading(HintFile(snapName))
+		if os.IsNotExist(err) {
+			if err := notInhibited(); err != nil {
+				return nil, err
+			}
+			return nil, nil
+		}
+		if err != nil {
+			return nil, err
+		}
+		// Release lock if we return early with an error
+		defer func() {
+			// Keep lock if no errors occur
+			if retErr != nil {
+				flock.Close()
+			}
+		}()
```

Comment:
this creates a defer per loop which is not great, it would be better to have this outside the loop possibly by using a predefined flock variable

Reply:
yes, this is definitely better! updated, thanks!